src/notification.rs

```diff
@@ -38,70 +41,66 @@ pub struct MustacheData {
     sites: Vec<Url>,
 }
 
-impl MustacheData {
-    pub fn new(commit: Option<String>, changed_urls: Vec<Url>) -> Self {
-        let mut sites = changed_urls;
-        sites.sort_unstable();
-        sites.dedup();
-
-        let mut hosts = sites
-            .iter()
-            .filter_map(Url::host_str)
-            .map(std::string::ToString::to_string)
-            .collect::<Vec<_>>();
-        hosts.dedup();
-
-        let singlehost = if let [single] = hosts.as_slice() {
+impl From<Summary> for MustacheData {
+    fn from(summary: Summary) -> Self {
+        let changed_hosts = summary.changed_hosts.into_keys().collect::<Vec<_>>();
+        let singlehost = if let [single] = changed_hosts.as_slice() {
             Some(single.clone())
         } else {
             None
         };
-
         Self {
-            commit,
+            commit: summary.commit,
             singledomain: singlehost.clone(),
             singlehost,
-            siteamount: sites.len(),
-
-            domains: hosts.clone(),
-            hosts,
-            sites,
+            siteamount: summary.changed_amount,
```

Comment:
Why is `siteamount` the same as `changed_amount`? My intuition with these is:
`site amount`: The number of sites that have changed
`changed_amount`: The number of changes on one site.

Reply:
`MustacheData` is legacy notification related stuff that is deprecated. This change ensures it's still working before it is removed. This means that the key names of `MustacheData` can not be changed. But the names on `Summary` are new and can be better than they were before.

`changed_amount` could be named `changed_site_amount`. I omitted the `site` part here as it felt redundant. Maybe adding it back is a good idea to be more clear about what it is.

Reply:
No need to be super duper perfect for already deprecated stuff. This explanation is good enough!